docs/core/v3.0.rst

```diff
@@ -0,0 +1,1761 @@
+    defined by an extension. If the chunk memory layout type
+    is defined by an extension, then the value must be an
+    object containing the names ``extension`` and ``type``. The
+    ``extension`` is required and the value must be a URI that
+    identifies the extension and dereferences to a
+    human-readable representation of the specification.  The ``type`` is
+    required and the value is defined by the extension.
+
+``fill_value``
+^^^^^^^^^^^^^^
+
+    Provides an element value to use for uninitialised portions of the
+    Zarr array.
+
+    If the data type of the Zarr array is Boolean then the value must
+    be the literal ``false`` or ``true``. If the data type is one of
+    the integer data types defined in this specification, then the
+    value must be a number with no fraction or exponent part and must
+    be within the range of the data type.
+
+    For any data type, if the ``fill_value`` is the literal ``null``
+    then the fill value is undefined and the implementation may use
+    any arbitrary value that is consistent with the data type as the
+    fill value.
+
+    If the ``data_type`` of an array is defined in a ``data_type`` extension,
+    then said extension is responsible for interpreting the value of
+    ``fill_value`` and return a suitable type that can be used.
+
+    For core data types for which fill values are not permitted in JSON or
+    for which decimal representation could be lossy, a string representing of
+    the binary (starting with ``0b``) or hexadecimal value (starting with
+    ``0x``) is accepted. This string must include all leading or trailing
+    zeroes necessary to match the given type size. The string values ``"NaN"``,
+    ``"+Infinity"`` and ``"-Infinity"`` are also understood for floating point
+    data types.
+
+``extensions``
+^^^^^^^^^^^^^^
+
+    See the top level metadata extension section for the time being.
+
+
+``attributes``
+^^^^^^^^^^^^^^
+
+    The value must be an object. The object may contain any name/value
+    pairs. Intended to allow storage of arbitrary user metadata
+
+
+  .. note::
+    The question of whether core metadata and user attributes should be
+    stored together or in separate documents is a topic of ongoing discussion.
+    (See https://github.com/zarr-developers/zarr-specs/issues/72.)
+
+
+The following names are optional:
+
+``compressor``
+^^^^^^^^^^^^^^
+
+    Specifies a codec to be used for encoding and decoding chunks. The
+    value must be an object containing the name ``codec`` whose value
+    is a URI that identifies a codec and dereferences to a human-readable
+    representation of the codec specification. The codec
+    object may also contain a ``configuration`` object which consists of the
+    parameter names and values as defined by the corresponding codec
+    specification. When the ``compressor`` name is absent, this means that no
+    compressor is used.
+
+``storage_transformers``
+^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
In line with https://github.com/zarr-developers/zarr-specs/pull/149/files#r950120916, I think we should also support storage transformers at the root level. This would become a `chunk_storage_transformer`.

Reply:
Tracked in #169 for the moment.